rpmlint/checks/SUIDPermissionsCheck.py

```diff
@@ -0,0 +1,212 @@
+            # return the new path first.
+            # chkstat prefers the new paths over the old ones, so callers that only care about the
+            # first matching file must mimic that.
+            yield '/usr/share/permissions/' + name
+            yield '/etc/' + name
+
+    def check(self, pkg):
+        if pkg.is_source:
+            return
+
+        need_set_permissions = False
+        found_suseconfig = False
+
+        for f, pkgfile in pkg.files.items():
+
+            if pkgfile.filecaps:
+                self.output.add_info(
+                    'E',
+                    pkg,
+                    'permissions-fscaps',
+                    "%(fname)s has fscaps '%(caps)s'"
+                    % {'fname': f, 'caps': pkgfile.filecaps},
+                )
+
+            mode = pkgfile.mode
+            owner = pkgfile.user + ':' + pkgfile.group
+
+            need_verifyscript = False
+            if f in self.perms or (stat.S_ISDIR(mode) and f + '/' in self.perms):
+                if stat.S_ISLNK(mode):
+                    self.output.add_info('W', pkg, 'permissions-symlink', f)
+                    continue
+
+                need_verifyscript = True
+
+                m = 0
+                o = 'invalid'
+                if stat.S_ISDIR(mode):
+                    if f in self.perms:
+                        self.output.add_info(
+                            'W', pkg, 'permissions-dir-without-slash', f
+                        )
+                    else:
+                        f += '/'
+
+                entry = self.perms[f]
+
+                if stat.S_ISREG(mode) and mode & (
+                    stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH
+                ):
+                    # pie binaries have 'shared object' here
+                    if (
+                        pkgfile.magic.startswith('ELF ') and ('shared object' not in pkgfile.magic) and ('pie executable' not in pkgfile.magic)
+                    ):
+                        self.output.add_info(
+                            'E', pkg, 'non-position-independent-executable', f
+                        )
+
+                m = entry.mode
+                o = ':'.join((entry.owner, entry.group))
+
+                if stat.S_IMODE(mode) != m:
+                    self.output.add_info(
+                        'E',
+                        pkg,
+                        'permissions-incorrect',
+                        '%(file)s has mode 0%(mode)o but should be 0%(m)o'
+                        % {'file': f, 'mode': stat.S_IMODE(mode), 'm': m},
+                    )
+
+                if owner != o:
+                    self.output.add_info(
+                        'E',
+                        pkg,
+                        'permissions-incorrect-owner',
+                        '%(file)s belongs to %(owner)s but should be %(o)s'
+                        % {'file': f, 'owner': owner, 'o': o},
+                    )
+
+            elif not stat.S_ISLNK(mode):
```

Comment:
Please also move the body of this `if` clause into a separate member function like `_checkRestrictedMode(self, path)`.